### PR TITLE
refactor(rubygems): Decouple metadata fetching from versions fetching

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

Decouple `getDependency` from metadata fetching method (`fetchGemMetadata`)

## Context

- Ref: #3670

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
Oops

**Review by viceice — COMMENTED** [review]
you need to update test http mocks

**Review by viceice — CHANGES_REQUESTED** [review]
wrong merge from main

File: lib/modules/datasource/rubygems/index.ts [hunk 1]
```diff
@@ -57,39 +57,42 @@ export class RubyGemsDatasource extends Datasource {
...
+      return null;
     } catch (error) {
       this.handleGenericErrors(error);
     }
-
-    return null;
   }
 
-  async getDependencyFallback(
+  async fetchGemMetadata(
```
Comment (viceice) [inline lib/modules/datasource/rubygems/index.ts:88]: why function reordering, makes PR more complicated than it has to

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/modules/datasource/rubygems/index.ts (hunk 1 above)
Comment (zharinov) [reply lib/modules/datasource/rubygems/index.ts:88]: Sorry for that, but new ordering is much better

Comment (viceice) [reply lib/modules/datasource/rubygems/index.ts:88]: ok, please do it in separate PR's next time.

**Review by viceice — APPROVED** [review]

**Pull request merged**